ext/standard/scanf.c

```diff
@@ -469,11 +469,11 @@ PHPAPI int ValidateFormat(char *format, int numVars, int *totalSubs)
 				}
 				break;
 badSet:
-				php_error_docref(NULL, E_WARNING, "Unmatched [ in format string");
+				zend_value_error("Unmatched [ in format string");
 				goto error;
 
 			default: {
-				php_error_docref(NULL, E_WARNING, "Bad scan conversion character \"%c\"", *ch);
+				zend_value_error("Bad scan conversion character \"%c\"", *ch);
 				goto error;
```

Comment:
ValidateFormat also includes a few other warnings -- any reason why only these two in particular were converted?

Reply:
I think it's because I was unsure how far I can go here. Originally, I wanted to do a followup PR if the these warning promotions are OK, but it's fine for me to promote them all in this PR.

Reply:
In that case I'd suggest to split off the full sscanf change as a separate PR.